lib/webdriveragent.js

```diff
@@ -190,17 +190,28 @@ class WebDriverAgent {
    *   }
    * }
    *
+   * @param {number} [timeoutMs=0] If the given timeoutMs is zero or negative number,
+   * this function will return the response of `/status` immediately. If the given timeoutMs,
+   * this function will try to get the response of `/status` up to the timeoutMs.
    * @return {Promise<any?>} State Object
    */
-  async getStatus () {
+  async getStatus (timeoutMs = 0) {
     const noSessionProxy = new NoSessionProxy({
       server: this.url.hostname,
       port: this.url.port,
       base: this.basePath,
       timeout: 3000,
     });
     try {
-      return await noSessionProxy.command('/status', 'GET');
+      if (timeoutMs <= 0) {
+        return await noSessionProxy.command('/status', 'GET');
+      }
+
+      return await retryInterval(
```

Comment:
I would still prefer waitForCondition here
retryInterval is not precise, since it does not count the duration of the /status request itself
just don't forget about exceptions handling

Reply:
oh. i see.

> it does not count the duration of the /status request itself

I initially used the waitForCondition but moved to the retryInterval. Let me move back to the waitForCondition